controllers/controller.go

```diff
@@ -325,6 +327,20 @@ func (r *EtcdadmClusterReconciler) reconcile(ctx context.Context, etcdCluster *e
 				delete(etcdCluster.Annotations, etcdv1.UpgradeInProgressAnnotation)
 			}
 		}
+
+		// If the ETCD nodes have performed a rollout, the etcd client certs on the CP nodes need to be renewed.
+		// We mark the condition EtcdCertificatesAvailable False in the rollout case and check for its value.
+		// The default case is hit right after ScaleUp of ETCD nodes is completed and before the first CP comes up.
+		// If EtcdCertificatesAvailable is False, this means we need to update the certs.
+		// EtcdCertificatesAvailable is set to True once the certs are updated.
+		if conditions.Has(ep.EC, etcdv1.EtcdCertificatesAvailableCondition) &&
```

Comment:
do we need `conditions.Has(ep.EC, etcdv1.EtcdCertificatesAvailableCondition)`? or `conditions.IsFalse(ep.EC, etcdv1.EtcdCertificatesAvailableCondition)` is sufficient?

Reply:
Um that might work. I followed the pattern on top line 322 where we checked `if conditions.Has(ep.EC, etcdv1.EtcdMachinesSpecUpToDateCondition)` before setting to True.

But it might work with just IsFalse since it returns False if the condition is not found.

Reply:
yea it will return false if condition is nil so i dont think you need the `Has` check